tests/floating-point/fft/fft-comb.fil

```diff
@@ -0,0 +1,178 @@
+import "primitives/signed.fil";
+import "primitives/float.fil";
+
+/// Toplevel FFT Module
+comp CombFFT[NStages, W, E, ?M=W-E-1]<'G: 1>(
+  in[NPoints*2]: ['G, 'G+1] W,
+) -> (
+  out[NPoints*2]: ['G, 'G+1] W
+) with {
+  let NPoints = pow2(NStages);
+} where
+  NStages > 0,
+  W == E + M + 1,
+  E > 1,
+  M > 1
+{
+  // generate twiddle factors
+  bundle twiddle[NPoints*2]: ['G, 'G+1] W;
+  for i in 0..pow2(NStages) {
+    cosbits := new SignedConst[W, cos_bits(i, NPoints)]<'G>();
+    sinbits := new SignedConst[W, sin_bits(i+pow2(NStages-1), NPoints)]<'G>(); // equivalent to -sin
+    twiddle{i*2} = cosbits.out;
+    twiddle{i*2+1} = sinbits.out;
+  }
+
+  // bit reverse the input
+  bundle bitrev[NPoints*2]: ['G, 'G+1] W;
+  for j in 0..NPoints {
+    let br = bit_rev(j, NStages);
+    assume br < NPoints;
```

Comment:
We should bring back post monomorphization checking for `assume` and `assert` statements. Stuff like this makes me a little nervous although seems obviously true.